src/scripts/main.js

```diff
@@ -1 +1,74 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  let resolved = false;
+
+  const timeout = setTimeout(() => {
+    if (!resolved) {
+      reject(new Error('First promise was rejected'));
+    }
+  }, 3000);
+
+  document.addEventListener('click', (e) => {
+    if (e.button === 0 && !resolved) {
+      clearTimeout(timeout);
+      resolved = true;
+      resolve('First promise was resolved');
+    }
```

Comment:
You are creating a new function every time the `click` event is triggered. This can lead to performance issues because each function consumes memory. It would be better to define this function outside of the event listener and reference it.